src/library/tabledelegates/stareditor.cpp

```diff
@@ -83,31 +89,132 @@ void StarEditor::paintEvent(QPaintEvent*) {
+            // Allow handling of certain keyboard navigation events.
+            // This logic attempts to match the behavior for text editor cells.
+            return false;
+        }
+        case Qt::Key_Home:
+        case Qt::Key_End: {
+            // Only forward Home and End if the Ctrl key is pressed.
+            // This matches the behavior of normal text editor cells.
+            if (ke->modifiers() & Qt::ControlModifier) {
+                return false;
+            }
+            return true;
+        }
+        case Qt::Key_0: {
+            newRating = 0;
+            break;
+        }
+        case Qt::Key_1: {
+            newRating = 1;
+            break;
+        }
+        case Qt::Key_2: {
+            newRating = 2;
+            break;
+        }
+        case Qt::Key_3: {
+            newRating = 3;
+            break;
+        }
+        case Qt::Key_4: {
+            newRating = 4;
+            break;
+        }
+        case Qt::Key_5: {
+            newRating = 5;
+            break;
+        }
+        case Qt::Key_6: {
+            newRating = 6;
+            break;
+        }
+        case Qt::Key_7: {
+            newRating = 7;
+            break;
+        }
+        case Qt::Key_8: {
+            newRating = 8;
+            break;
+        }
+        case Qt::Key_9: {
+            newRating = 9;
+            break;
+        }
+        case Qt::Key_Right:
+        case Qt::Key_Plus: {
+            newRating++;
+            break;
+        }
+        case Qt::Key_Left:
+        case Qt::Key_Minus: {
+            newRating--;
+            break;
+        }
+        }
+        newRating = math_clamp(newRating, StarRating::kMinStarCount, m_starRating.maxStarCount());
+        if (newRating != m_starRating.starCount()) {
+            // Apply star rating if it changed
+            m_starRating.setStarCount(newRating);
+            m_starCount = newRating;
+            update();
+        }
+        // Prevent other keys from being handled as global keyboard shortcuts.
+        // This matches the behavior of the other edit controls.
+        return true;
+    }
     case QEvent::Leave:
     case QEvent::ContextMenu: {
         // Note: it seems with Qt5 we do not reliably get a Leave event when
         // invoking the track menu via right click, so reset the rating now.
         // The event is forwarded to parent QTableView.
+        m_starCountToSave = StarRating::kInvalidStarCount;
         resetRating();
         break;
     }
+    case QEvent::MouseButtonPress: {
+        // Workaround: The MouseButtonPress can cause a focus change that might
+        // cause a database reload, which then overwrites m_starRating before
+        // the MouseButtonRelease event has a chance to commit the new value
+        // to the model.
+        m_starCountToSave = m_starRating.starCount();
+        break;
+    }
     case QEvent::MouseButtonRelease: {
+        if (m_starCountToSave != StarRating::kInvalidStarCount) {
```

Comment:
Just set the rating, `setStarCount` does `verifyStarCount(int)`

Reply:
`setStarCount` will trigger a debug assertion on `kInvalidStarCount`, but `m_starCountToSave == kInvalidStarCount` does not indicate a bug, but is a valid state here.